Source/Core/DolphinQt2/TAS/WiiTASInputWindow.cpp

```diff
@@ -0,0 +1,431 @@
+  classic_buttons_layout2->addWidget(m_classic_down_button);
+  classic_buttons_layout2->addWidget(m_classic_right_button);
+
+  auto* classic_buttons_layout = new QVBoxLayout;
+  classic_buttons_layout->setSizeConstraint(QLayout::SetFixedSize);
+  classic_buttons_layout->addLayout(classic_buttons_layout1);
+  classic_buttons_layout->addLayout(classic_buttons_layout2);
+
+  m_classic_buttons_box = new QGroupBox(tr("Classic Buttons"));
+  m_classic_buttons_box->setLayout(classic_buttons_layout);
+
+  auto* layout = new QVBoxLayout;
+  layout->addLayout(top_layout);
+  layout->addWidget(m_remote_orientation_box);
+  layout->addWidget(m_nunchuk_orientation_box);
+  layout->addWidget(m_triggers_box);
+  layout->addWidget(m_remote_buttons_box);
+  layout->addWidget(m_nunchuk_buttons_box);
+  layout->addWidget(m_classic_buttons_box);
+  layout->setAlignment(m_nunchuk_buttons_box, Qt::AlignLeft);
+
+  setLayout(layout);
+
+  u8 ext = 0;
+  if (Core::IsRunning())
+  {
+    ext = static_cast<WiimoteEmu::Wiimote*>(Wiimote::GetConfig()->GetController(num))
+              ->CurrentExtension();
+  }
+  else
+  {
+    IniFile ini;
+    ini.Load(File::GetUserPath(D_CONFIG_IDX) + "WiimoteNew.ini");
+    std::string extension;
+    ini.GetIfExists("Wiimote" + std::to_string(num + 1), "Extension", &extension);
+
+    if (extension == "Nunchuk")
+      ext = 1;
+    if (extension == "Classic")
+      ext = 2;
+  }
+  UpdateExt(ext);
+}
+
+void WiiTASInputWindow::UpdateExt(u8 ext)
+{
+  if (ext == 1)
+  {
+    setWindowTitle(tr("Wii TAS Input %1 - Remote + Nunchuk").arg(m_num + 1));
+    m_ir_box->show();
+    m_nunchuk_stick_box->show();
+    m_classic_right_stick_box->hide();
+    m_classic_left_stick_box->hide();
+    m_remote_orientation_box->show();
+    m_nunchuk_orientation_box->show();
+    m_triggers_box->hide();
+    m_nunchuk_buttons_box->show();
+    m_remote_buttons_box->show();
+    m_classic_buttons_box->hide();
+  }
+  else if (ext == 2)
+  {
+    setWindowTitle(tr("Wii TAS Input %1 - Classic Controller").arg(m_num + 1));
+    m_ir_box->hide();
+    m_nunchuk_stick_box->hide();
+    m_classic_right_stick_box->show();
+    m_classic_left_stick_box->show();
+    m_remote_orientation_box->hide();
+    m_nunchuk_orientation_box->hide();
+    m_triggers_box->show();
+    m_remote_buttons_box->hide();
+    m_nunchuk_buttons_box->hide();
+    m_classic_buttons_box->show();
+  }
+  else
+  {
+    setWindowTitle(tr("Wii TAS Input %1 - Remote").arg(m_num + 1));
+    m_ir_box->show();
+    m_nunchuk_stick_box->hide();
+    m_classic_right_stick_box->hide();
+    m_classic_left_stick_box->hide();
+    m_remote_orientation_box->show();
+    m_nunchuk_orientation_box->hide();
+    m_triggers_box->hide();
+    m_remote_buttons_box->show();
+    m_nunchuk_buttons_box->hide();
+    m_classic_buttons_box->hide();
+  }
+}
+
+template <typename ux> static void SetButton(QCheckBox* check_box, ux* buttons, ux mask)
```

Comment:
Our convention with templates is to place the function underneath the template declarator like:

```cpp
template <typename ux>
static void SetButton(QCheckBox* check_box, ux* buttons, ux mask)
```

template parameters are usually upper-cased as well.